raft: TestNodeStepUnblock failed with data race

```
=== RUN   TestNodeStepUnblock
==================
WARNING: DATA RACE
Write by goroutine 8:
  github.com/coreos/etcd/raft.TestNodeStepUnblock()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node_test.go:95 +0x73f
  testing.tRunner()
      /home/suda/go/src/testing/testing.go:473 +0xdc

Previous read by goroutine 62:
  github.com/coreos/etcd/raft.TestNodeStepUnblock.func2()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node_test.go:84 +0x5e

Goroutine 8 (running) created at:
  testing.RunTests()
      /home/suda/go/src/testing/testing.go:582 +0xae2
  testing.(*M).Run()
      /home/suda/go/src/testing/testing.go:515 +0x11d
  main.main()
      github.com/coreos/etcd/raft/_test/_testmain.go:368 +0x210

Goroutine 62 (finished) created at:
  github.com/coreos/etcd/raft.TestNodeStepUnblock()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node_test.go:86 +0x561
  testing.tRunner()
      /home/suda/go/src/testing/testing.go:473 +0xdc
==================
==================
WARNING: DATA RACE
Write by goroutine 8:
  github.com/coreos/etcd/raft.TestNodeStepUnblock()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node_test.go:99 +0x7e8
  testing.tRunner()
      /home/suda/go/src/testing/testing.go:473 +0xdc

Previous read by goroutine 62:
  github.com/coreos/etcd/raft.(*node).step()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:423 +0xda
  github.com/coreos/etcd/raft.(*node).Step()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:399 +0xde
  github.com/coreos/etcd/raft.TestNodeStepUnblock.func2()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node_test.go:84 +0x9c

Goroutine 8 (running) created at:
  testing.RunTests()
      /home/suda/go/src/testing/testing.go:582 +0xae2
  testing.(*M).Run()
      /home/suda/go/src/testing/testing.go:515 +0x11d
  main.main()
      github.com/coreos/etcd/raft/_test/_testmain.go:368 +0x210

Goroutine 62 (finished) created at:
  github.com/coreos/etcd/raft.TestNodeStepUnblock()
      /home/suda/WORK/etcd/gopath/src/github.com/coreos/etcd/raft/node_test.go:86 +0x561
  testing.tRunner()
      /home/suda/go/src/testing/testing.go:473 +0xdc
==================
--- FAIL: TestNodeStepUnblock (0.11s)
        node_test.go:103: #0: failed to unblock step
        node_test.go:91: #1: err = raft: stopped, want context canceled
```

Tested version: f13eaac7c38b1348f6eb2cc44218c9c857074fd1
Note that the reproducibility is very low.

The cause seems to be a race in the test code, and hence it does not affect production.

--- a/raft/node_test.go
+++ b/raft/node_test.go
@@ -99,8 +99,8 @@ func TestNodeStepUnblock(t *testing.T) {
 				n.done = make(chan struct{})
 			default:
 			}
-		case <-time.After(time.Millisecond * 100):
-			t.Errorf("#%d: failed to unblock step", i)
+		case <-time.After(1 * time.Second):
+			t.Fatalf("#%d: failed to unblock step", i)
 		}
 	}
 }

